isis/src/docsys/documents/DeveloperAddFeature/DeveloperAddFeature.xml

```diff
@@ -0,0 +1,187 @@
+              it on the ISIS GitHub site. Doing this will allow other ISIS developers to way in on your preposed feature. By posting
+              a description of what you want to do and asking questions, other developers can give you pointers on how to 
+              implement it.
+            </p>
+
+          <h2><a name="Writing your new application">Writing your new application</a></h2>
+
+            <h3>Creating the files for your new application</h3>
+            <p>
+              The ISIS source code is organized by having a dedicated directory where all of the source code for an application is
+              stored. The files in this directory will be exclusively to your feature. This is where using another application as
+              a template can be very useful.
+            </p>
+            <ol>
+                <li>Decide where your new application belongs</li>
+                <li>Create a directory using the same name as you new application</li>
+                <li>Create a "main.cpp" file</li>
+                <li>Create a "your_new_app.cpp" file</li>
+                <li>Create a "your_new_app.xml" file</li>
+                <li>Create any additional support files (e.g., support_class.cpp, support_class.h</li>
+            </ol>
+            <p>
+              Populate the new files with the code necessary to implement your feature.
+            </p>
+
+            <h3>Compile and link your new application for the first time</h3>
+            <p>
+              Since you are adding a new application to ISIS you will need to re-run the cmake command. This will add 
+              the new application to the list of applications the build system knows about.
+            </p>
+            <ol>
+              <li>Change you current working directory to the ISIS build directory (cd $ISISROOT)</li>
+              <li>Remove the contents of the build directory (rm -rf *)</li>
+              <li>Re-run the cmake command (cmake -GNinja ..)</li>
+              <li>Build the ISIS library and all of the applications (ninja)</li>
+            </ol>
+
+            <h3>Fix, run, hand test, debug the new application</h3>
+            <p>
+              Iterate over the following steps until you have your new feature producing the desired result 
+            </p>
+            <ol>
+              <li>Fix any compile or linking errors</li>
+              <li>Rebuild the ISIS library and your new application (ninja)</li>
+              <li>Run your application and test the user interface</li>
+              <li>Run your application and test the output</li>
+                <ol>
+                  <li>Cube DNs</li>
+                  <li>Cube labels</li>
+                  <li>History</li>
+                  <li>Original label</li>
+                  <li>print.prt file</li>
+                  <li>Cleanup any temporary files</li>
```

Comment:
What does this list mean? What should developers be checking for here? Being more specific like "Does the app add a history entry to all output cubes?"